lib/scrub/cip/connection_manager.ex

```diff
@@ -114,8 +114,22 @@ defmodule Scrub.CIP.ConnectionManager do
   end
 
   def encode_service(:unconnected_send, opts) do
-    request_path = opts[:request_path]
+    request_path = encode_request_path(opts[:request_path])
 
+    request_words = (byte_size(request_path) / 2) |> floor
+
+    read_elements = opts[:read_elements] || 1
```

Comment:
Not blocking.

In general it is suboptimal to use `map_or_keyword[key] || default` because you will get the default value even if `map_or_keyword` has a value for the requested key if the value is explicitly `nil` or `false`. Better to use `Map/Keyword.get(map_or_keyword, ket, default`. You protect yourself from future regressions and other developers don't have to trust or verify what might be in `opts` to understand what will happen here.

Reply:
Im pretty sure this command fails if this is not 1 or more. So if nil or false is passed in, it MUST be converted to the default of 1. Perhaps we need more guards around this to also specify that is must be an integer of 1 or greater?

Reply:
If you feel protected here you're good :). It is generally suboptimal, not always!